src/Validator/ResolveValidator/ResolveValidator.php

```diff
@@ -174,7 +175,12 @@ public function assertTypeInUnionTypes(AbstractType $type, AbstractUnionType $un
      */
     public function assertValidFragmentForField(Fragment $fragment, FragmentReference $fragmentReference, AbstractType $queryType)
     {
-        if ($fragment->getModel() !== $queryType->getName()) {
+        $innerType = $queryType;
+        while ($innerType->isCompositeType()) {
```

Comment:
It would mean that you are basically recursively nesting composite types which makes zero sense.